### ISSUE TITLE
VXLAN as isolation method when using Advanced networks with security groups is ignored when deploying VMs

### ISSUE THREAD

**Issue by PaulAngus** [apache/cloudstack#2701]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Networking/VM deployment
~~~

##### CLOUDSTACK VERSION

~~~
4.11.1 and before
~~~

##### CONFIGURATION

VXLAN as isolation method when using Advanced networks with security groups

##### SUMMARY

Set up a zone with VXLAN as isolation method when using Advanced networks with security groups

Attempt to deploy a VM

##### STEPS TO REPRODUCE

~~~
Set up a zone with VXLAN as isolation method when using Advanced networks with security groups

![image](https://user-images.githubusercontent.com/4810220/41107358-1b5ec2aa-6a6a-11e8-873c-d7f03f677078.png)
In this case vxlan id 10000 was used for the default guest network

Attempt to deploy a VM

~~~

##### EXPECTED RESULTS

~~~

CloudStack creates a network with VXLAN ID 10000 for virtual router.

~~~

##### ACTUAL RESULTS

~~~

CloudStack attempts to create a VLAN with ID 10000
which causes the following error: 
[note: type=vlan]

2018-06-07 14:26:55,518 DEBUG [c.c.a.t.Request] (AgentManager-Handler-10:null) (logid:) Seq 1-8751619975887716389: Processing:  { Ans: , MgmtId: 6738937972795, via: 1, Ver: v1, Flags: 10, [{"com.cloud.agent.api.StartAnswer":{"vm":{"id":4,"name":"r-4-VM","type":"DomainRouter","cpus":1,"minSpeed":500,"maxSpeed":500,"minRam":268435456,"maxRam":268435456,"arch":"x86_64","os":"Debian GNU/Linux 5.0 (64-bit)","platformEmulator":"Debian GNU/Linux 5","bootArgs":" template=domP name=r-4-VM eth0ip=192.168.99.100 eth0mask=255.255.255.0 gateway=192.168.99.1 domain=cs1cloud.internal cidrsize=24 dhcprange=192.168.99.1 eth1ip=169.254.2.60 eth1mask=255.255.0.0 type=dhcpsrvr disable_rp_filter=true dns1=8.8.8.8 baremetalnotificationsecuritykey=Puf4MwH7oeihrUcVjiNGLaMIitnqmMvZ-kjUcE3_iXFW74BOUlD95CyAoxLMzBFQ8FZuVm-qUn2zVv5XG-Qn4A baremetalnotificationapikey=2kvJ3-hhENZC3HYYUsUm09dLVP66UeCjFB5I7JoTs-D51HzCTvqvSAY23sofRGJLHxYQ4K21Q4T-U9AYZh5hmA host=10.2.3.67 port=8080","enableHA":true,"limitCpuUse":false,"enableDynamicallyScaleVm":false,"vncPassword":"5C6RNBlMMOpi917i61mVnA","vncAddr":"10.2.3.207","params":{},"uuid":"a819cebe-828f-4f85-a659-9806cf13d6dc","disks":[{"data":{"org.apache.cloudstack.storage.to.VolumeObjectTO":{"uuid":"032093b3-6c75-4ee4-ad84-3f2212c0cfbf","volumeType":"ROOT","dataStore":{"org.apache.cloudstack.storage.to.PrimaryDataStoreTO":{"uuid":"a9a665a2-ac37-3621-9d84-fb96af17877a","id":1,"poolType":"NetworkFilesystem","host":"10.2.0.16","path":"/acs/primary/pr2693-t2738-kvm-centos7/pr2693-t2738-kvm-centos7-kvm-pri1","port":2049,"url":"NetworkFilesystem://10.2.0.16/acs/primary/pr2693-t2738-kvm-centos7/pr2693-t2738-kvm-centos7-kvm-pri1/?ROLE=Primary&STOREUUID=a9a665a2-ac37-3621-9d84-fb96af17877a","isManaged":false}},"name":"ROOT-4","size":305704448,"path":"032093b3-6c75-4ee4-ad84-3f2212c0cfbf","volumeId":4,"vmName":"r-4-VM","accountId":1,"format":"QCOW2","provisioningType":"THIN","id":4,"deviceId":0,"bytesReadRate":0,"bytesWriteRate":0,"iopsReadRate":0,"iopsWriteRate":0,"hypervisorType":"KVM"}},"diskSeq":0,"path":"032093b3-6c75-4ee4-ad84-3f2212c0cfbf","type":"ROOT","_details":{"storageHost":"10.2.0.16","managed":"false","storagePort":"2049","volumeSize":"305704448"}}],"nics":[{"deviceId":0,"networkRateMbps":200,"defaultNic":true,"pxeDisable":true,"nicUuid":"15de3db8-b329-4e8e-84dd-68bf3c926304","details":{"PromiscuousMode":"false","MacAddressChanges":"true","ForgedTransmits":"true"},"uuid":"705beff1-abbf-40c2-842a-ebb3f853da42","ip":"192.168.99.100","netmask":"255.255.255.0","gateway":"192.168.99.1","mac":"1e:00:72:00:00:29","dns1":"8.8.8.8","broadcastType":"Vlan","type":"Guest","broadcastUri":"vlan://10000","isolationUri":"vlan://10000","isSecurityGroupEnabled":true,"name":"cloudbr1"},{"deviceId":1,"networkRateMbps":-1,"defaultNic":false,"pxeDisable":true,"nicUuid":"7e40f313-6df0-4071-a41a-9463c5d20e99","details":{"PromiscuousMode":"false","MacAddressChanges":"true","ForgedTransmits":"true"},"uuid":"1e448126-cead-4b44-8b5d-01bffa97f0e3","ip":"169.254.2.60","netmask":"255.255.0.0","gateway":"169.254.0.1","mac":"0e:00:a9:fe:02:3c","broadcastType":"LinkLocal","type":"Control","isSecurityGroupEnabled":false}],"guestOsDetails":{}},"result":false,"details":"Failed to create vnet 10000: RTNETLINK answers: Numerical result out of rangeCannot find device \"eth1.10000\"Failed to create vlan 10000 on pif: eth1.","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}},{"com.cloud.agent.api.Answer":{"result":false,"details":"Stopped by previous failure","wait":0}}] }

~~~

**Comment by nvazquez** [comment]
@rhtyd yes, it does

**Comment by rhtyd** [comment]
Closing, fixed in #2693
